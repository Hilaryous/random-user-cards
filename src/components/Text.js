// @flow
import styled from 'react-emotion'

const Text = styled('p')({
  fontSize: 13,
  color: '#294762',
  lineHeight: 1.4,
  margin: 0,
},
({
  modifier,
}) => {
  switch (modifier) {
    case 'italic':
      return { fontStyle: 'italic' }
    case 'bold':
      return { fontWeight: 'bold' }
    case 'center':
      return { textAlign: 'center' }
    case 'post':
      return {
        opacity: '0.8',
        lineHeight: '1.5',
      }
    default:
      return {}
  }
})

export default Text
