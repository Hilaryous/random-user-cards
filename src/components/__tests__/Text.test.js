/* eslint-disable function-paren-newline */
import React from 'react'
import renderer from 'react-test-renderer'
import Text from '../Text'

describe('Text', () => {
  it('shows the correct snapshot tree for the Text component', () => {
    const tree = renderer.create(
      <Text />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the Text component, italic', () => {
    const tree = renderer.create(
      <Text
        modifier="italic"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the Text component, bold', () => {
    const tree = renderer.create(
      <Text
        modifier="bold"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the Text component, center', () => {
    const tree = renderer.create(
      <Text
        modifier="center"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the Text component, post', () => {
    const tree = renderer.create(
      <Text
        modifier="post"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
