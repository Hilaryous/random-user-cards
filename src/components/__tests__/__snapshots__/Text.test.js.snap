// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text shows the correct snapshot tree for the Text component 1`] = `
<p
  className="css-tku0cv"
/>
`;

exports[`Text shows the correct snapshot tree for the Text component, bold 1`] = `
<p
  className="css-11losmg"
/>
`;

exports[`Text shows the correct snapshot tree for the Text component, center 1`] = `
<p
  className="css-19nqlgd"
/>
`;

exports[`Text shows the correct snapshot tree for the Text component, italic 1`] = `
<p
  className="css-18fexks"
/>
`;

exports[`Text shows the correct snapshot tree for the Text component, post 1`] = `
<p
  className="css-1itr8po"
/>
`;
